templates/templates.go

```diff
@@ -0,0 +1,383 @@
+			return template.HTML(output.String())
+		},
+		"parseUsersIntoHTML": func(tmplUsers []user.StrippedUser, userID int) template.HTML {
+			memberAdmin := t.permissionsParser(userID, permissions.ManageMembersMembersAdmin.String())
+			var output strings.Builder
+			for _, tmplUser := range tmplUsers {
+				var enabled, deleted, ifView string
+				if tmplUser.Enabled {
+					enabled = "Enabled"
+				} else {
+					enabled = "Disabled"
+				}
+				if tmplUser.Deleted {
+					deleted = "Deleted"
+				} else {
+					deleted = "-"
+				}
+				if memberAdmin {
+					ifView = fmt.Sprintf("<td><a href=\"/internal/user/%d\">View</a></td>", tmplUser.UserID)
+				}
+				output.WriteString(fmt.Sprintf("<tr><td>%d</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td>%s</tr>", tmplUser.UserID, tmplUser.Name, tmplUser.Username, tmplUser.Email, enabled, deleted, tmplUser.LastLogin, ifView))
+			}
+			return template.HTML(output.String())
+		},
+		"parseUserIntoHTML": func(u user.DetailedUser, userID int) template.HTML {
+			permissionAdmin := t.permissionsParser(userID, permissions.ManageMembersPermissions.String())
+			roleAdmin := t.permissionsParser(userID, permissions.ManageMembersGroup.String())
+			var output, perms, roles strings.Builder
+			var deleted, enabled, ldap, avatar, lastLogin, created, updated, deletedBy string
+			if len(u.Permissions) > 0 {
+				perms.WriteString("Permissions: <ol>")
+				for _, p := range u.Permissions {
+					if permissionAdmin {
+						perms.WriteString(fmt.Sprintf("<li style='list-style-type: none;'><span class='tab'></span><a href='/internal/permission/%d'>%s</a></li>", p.PermissionID, p.Name))
+					} else {
+						perms.WriteString(fmt.Sprintf("<li style='list-style-type: none;'><span class='tab'></span>%s</li>", p.Name))
+					}
+				}
+				perms.WriteString("</ol><br>")
+			}
+			if len(u.Roles) > 0 {
+				roles.WriteString("Roles: <ol>")
+				for _, r := range u.Roles {
+					if roleAdmin {
+						roles.WriteString(fmt.Sprintf("<li style='list-style-type: none;'><span class='tab'></span><a href='/internal/role/%d'>%s</a></li>", r.RoleID, r.Name))
+					} else {
+						roles.WriteString(fmt.Sprintf("<li style='list-style-type: none;'><span class='tab'></span>%s</li>", r.Name))
+					}
+				}
+				roles.WriteString("</ol><br>")
+			}
+			if u.DeletedBy.UserID != -1 {
+				deleted = "<h2 class='subtitle'><strong>Deleted!</strong></h2>"
+			}
+			if u.Enabled {
+				enabled = "enabled"
+			} else {
+				enabled = "disabled"
+			}
+			if u.LDAPUsername.Valid {
+				ldap = fmt.Sprintf("LDAP (Active Directory) username: %s<br>", u.LDAPUsername.String)
+			}
+			if u.UseGravatar {
+				avatar = "Using gravatar"
+			} else {
+				avatar = "Using local file"
+			}
+			if u.LastLogin.Valid {
+				lastLogin = fmt.Sprintf("Last login at %s<br>", u.LastLogin.String)
```

Comment:
```suggestion
				lastLogin = fmt.Sprintf("Last login at %s<br>", u.LastLogin.String())
```

Reply:
No that won't work, it is a `String` not `String()` as it is from the class `null.String`, handles `null` values from a database but does prove useful later on and in templates so it stuck, there is no method `String()` but I'm pulling the `String` from the `struct`

Reply:
Ah yeah you're right